### ISSUE TITLE
Signal breaks unless-stopped restart policy

### ISSUE THREAD

**Issue by leonshaw** [moby/moby#41302]
**Description**

Sending non-fatal signal to a container with `unless-stopped` restart policy prevents it from restarting.

**Steps to reproduce the issue:**

```
docker run -d --name=test --restart=unless-stopped nginx
docker kill -s HUP test
systemctl restart docker
```

**Describe the results you received:**

The container won't restart.

**Describe the results you expected:**

The container should restart.

**Additional information you deem important (e.g. issue happens only occasionally):**

#26464 fixed #11065 for `always` restarting containers, but not for `unless-stopped`. The container is still marked as `HasBeenManuallyStopped`.

**Output of `docker version`:**

```
Client:
 Version:           19.03.12-ce
 API version:       1.40
 Go version:        go1.14.5
 Git commit:        48a66213fe
 Built:             Sat Jul 18 01:33:21 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          19.03.12-ce
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.14.5
  Git commit:       48a66213fe
  Built:            Sat Jul 18 01:32:59 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.3.4.m
  GitCommit:        d76c121f76a5fc8a462dc64594aea72fe18e1178.m
 runc:
  Version:          1.0.0-rc91
  GitCommit:        24a3cf88a7ae5f4995f6750654c0e2ca61ef4bb2
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Comment by thaJeztah** [comment]
Thanks for reporting; I recall some other issues being reported in (what I think) is the same area, so looks like something that needs to be looked into. (Contributions welcome 😅). Some (possibly) related issues; https://github.com/moby/moby/issues/37137, https://github.com/moby/moby/issues/38954

**Comment by leonshaw** [comment]
Should define the set of signals that are considered terminating the container when `StopSignal` is or is not specified.
Or probably add a new attribute (like `FatalSignals`) to images or containers?

**Comment by thaJeztah** [comment]
I don't think we should complicate things too much by adding a `FatalSignals` property (that said; there's will be some corner-cases where non-stop signals may be handled by the container's process as "stop", such as [`SIGWINCH` for Apache](https://bz.apache.org/bugzilla/show_bug.cgi?id=50669), but yes, the "normal" stop-signal(s) and the stop-signal defined by the container should be handled correctly.

**Comment by dannmartens** [comment]
Instead of using the Docker CLI, you can use the native kill utility to send a signal to the container process. This will not affect the restart policy of the container.

This [script ](https://gist.github.com/dannmartens/1591603515195ef4f6f23c13ce3fdb24) automates the process of retrieving the process ID from the container ID you pass as an argument:
`$ dkill sigusr1 MY_CONTAINER_ID` 

When the container process is not owned by the user who runs the script, you will need elevated privileges. 

When possible, run the container using `--user $(id -u):$(id -g)` to avoid the need for privilege elevation.

**Comment by adarshpets** [comment]
Hello, I would like to contribute to this project! I am a student at the University of Michigan, and one of our assignments is to help contribute to a project, I would love to try and solve the issue, please let me know any other details to solve this!
